### PR TITLE
closes #987: use SortedMap.keySet instead of SortedMap.keys

### LINKED ISSUE #987: [BUG] Heisenbug when applying EXCEPT to records
Consider the following spec:

```tla
--------------------------------- MODULE RecordExcept ------------------------
EXTENDS Integers

VARIABLES
    (*
      @typeAlias: R = [
         id: Str,
         name: Str,
         oneMoreField: Bool,
         age: Int,
         isShadow: Bool,
         isAllow: Bool
      ];
      @type: Set(R);
    *)
    recs,
    \* @type: Bool;
    ok

Init ==
  /\ ok = TRUE
  /\ recs = { }

Next ==
  /\ ok' = \A r \in recs:
        r.age < 15
  /\ \/ LET nr == [name |-> "foo",
                   age |-> 10,
                   isShadow |-> TRUE,
                   id |-> "111",
                   isAllow |-> FALSE,
                   oneMoreField |-> TRUE
                   ]
       IN
       recs' = recs \union { nr }
    \/ \E r \in recs:
        LET nr == [r EXCEPT !.age = @ + 1] IN
        /\ recs' = (recs \union {nr}) \ {r}
        /\ nr /= r
    \/ UNCHANGED recs

Inv ==
    \A r \in recs:
        r.age < 15

=============================================================================
```

The model checker is failing on it with an obscure exception:

```
$ ~/devl/apalache/bin/apalache-mc check --inv=Inv RecordExcept.tla
...
State 3: Checking 1 state invariants                              I@10:33:50.204
(Please report an issue at: [https://github.com/informalsystems/apalache/issues],at.forsyte.apalache.tla.lir.MalformedTlaError: Checking values of incomparable types for equality: Const and Int)
<unknown>: unexpected TLA+ expression: Checking values of incomparable types for equality: Const and Int E@10:33:50.228
at.forsyte.apalache.tla.lir.MalformedTlaError: Checking values of incomparable types for equality: Const and Int
	at at.forsyte.apalache.tla.bmcmt.LazyEquality.cacheOneEqConstraint(LazyEquality.scala:130)
	at at.forsyte.apalache.tla.bmcmt.LazyEquality.keyEq$1(LazyEquality.scala:445)
	at at.forsyte.apalache.tla.bmcmt.LazyEquality.$anonfun$mkRecordEq$1(LazyEquality.scala:462)
	at scala.collection.immutable.List.map(List.scala:293)
	at at.forsyte.apalache.tla.bmcmt.LazyEquality.mkRecordEq(LazyEquality.scala:462)
...
```

This is the bug that have been annoying us for long. We have tracked it down to:

https://github.com/informalsystems/apalache/blob/74edba834a388a38f090baae64a4b34d650b3f5c/tla-bmcmt/src/main/scala/at/forsyte/apalache/tla/bmcmt/rules/FunExceptRule.scala#L146-L148

The explanation: we are using `SortedMap.keys` instead of `SortedMap.keySet`. Who would have guessed that `SortedMap.keys` are not guaranteed to be sorted. Luckily, the rest of the codebase is using `SortedMap.keySet`.

### PR DESCRIPTION
**Pull request by konnov**
closes #987

- [x] Tests added for any new code
- [x] Ran `make fmt-fix` (or had formatting run automatically on all files edited)
- [ ] ~Documentation added for any new functionality~
- [x] Entry added to [UNRELEASED.md](./UNRELEASED.md) for any new functionality

### CODE REVIEW

**Review by Kukovec — APPROVED** [review]
LGTM

File: UNRELEASED.md [hunk 1]
```diff
@@ -10,3 +10,7 @@
          * Some bug fix, see #124
 
      DO NOT LEAVE A BLANK LINE BELOW THIS PREAMBLE -->
+
```
Comment (shonfeder) [inline UNRELEASED.md:13]:
```suggestion
```

File: UNRELEASED.md [hunk 2]
```diff
@@ -10,3 +10,7 @@
...
+### Bug fixes
+
+* Heisenbug caused by EXCEPT on records, which used unsorted keys, see #987
```
Comment (shonfeder) [inline UNRELEASED.md:16]:
```suggestion
* Fixed a heisenbug caused by EXCEPT on records, which used unsorted keys, see #987
```

File: test/tla/cli-integration-tests.md [hunk 3]
```diff
@@ -1050,21 +1050,30 @@ EXITCODE: OK
...
+EXITCODE: OK
+```
+
 ## configure the check command
 
-Testing various flags that are set via command-line options and the TLC
-configuration file. The CLI has priority over the TLC config. So we have to
-test that it all works together.
+Testing various flags that are set via command-line options and the TLC configuration file. The CLI has priority over
+the TLC config. So we have to test that it all works together.
```
Comment (shonfeder) [inline test/tla/cli-integration-tests.md:1076]: Any idea what's up with these formatting changes? The shorter wrapped lines are nicer imo,

**Review by shonfeder — APPROVED** [review]

File: test/tla/cli-integration-tests.md (hunk 3 above)
Comment (konnov) [reply test/tla/cli-integration-tests.md:1076]: True. Don't know what happened

**Pull request merged**